src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/ShellContextExtensions.cs

```diff
@@ -15,12 +15,10 @@ public static class ShellContextExtensions
         /// </summary>
         public static Task<(ILocker locker, bool locked)> TryAcquireShellActivateLockAsync(this ShellContext shellContext)
         {
-            // If the shell is initializing, force the usage of a local lock.
-            var lockService = shellContext.Settings.State == TenantState.Initializing
-                ? (ILock)shellContext.ServiceProvider.GetRequiredService<ILocalLock>()
-                : shellContext.ServiceProvider.GetRequiredService<IDistributedLock>();
+            var lockService = shellContext.ServiceProvider.GetRequiredService<IDistributedLock>();
 
```

Comment:
Revert the change of this class, i explained above why i did it, yes maybe good to not check `Initializing` but for now a setup is not intended to be run concurrently, and if so we just let it fail.

Then, we want to be able to run the AutoSetup concurrently, but this is at another layer that we will allow this, by using the RedisLock in the AutoSetup middleware.

So we can revert the changes of `ShellContextExtensions`.

Reply:
Ok, will revert those changes, I tested this part it always resolves ILocalLock -> LocalLock even if RedisLock Feature enabled, meaning that part `shellContext.ServiceProvider.GetRequiredService<IDistributedLock>();` will never be used unless I am  missing something

Reply:
First, if necessary i will have to try it but for now i let you try it with your existing configuration, so for now on my side it's only theorical ;)

So, if you add the RedisLock to the setup features it doesn't mean that when activating the internal shell, that we built for the setup, includes this feature.

Hmm, even if you add the RedisLock feature to your setup recipe, the first time the shell is activated and will execute the Features recipe step, the RedisLock feature is not yet enable.

The important thing to check is, when adding the Redislock to the setup features, that we resolve the RedisLock in the auto setup middleware, is it the case?